qcore/asserts.py

```diff
@@ -353,6 +362,51 @@ def __exit__(self, exc_type, exc_val, exc_tb):
         raise AssertionError(message)
 
 
+class AssertRaisesInstance(object):
+    """With-context that asserts that the code within the context raises a specified exception
+    instance."""
+
+    def __init__(self, *expected_exception_instances, **kwargs):
+        assert (
+            len(expected_exception_instances) >= 1
+        ), "You must specify an exception instance when using AssertRaisesInstance"
+        self.expected_exception_instances = expected_exception_instances
+        self.extra = kwargs.pop("extra", None)
+        assert_eq({}, kwargs)
+
+    def __enter__(self):
+        self.ctx = AssertRaises(
```

Comment:
Why do you need to go through an underlying AssertRaises context? Seems like we should be able to just check the instances directly.

Reply:
Good point, does not really save much to use that. Fixed to not use it.